frontend/src/components/DateRangePicker/components/DateRangePickerInput.tsx

```diff
@@ -0,0 +1,81 @@
+import ReactInputMask from 'react-input-mask'
+import { Flex, forwardRef, Stack, Text, useMergeRefs } from '@chakra-ui/react'
+
+import Input from '~components/Input'
+
+import { useDateRangePicker } from '../DateRangePickerContext'
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+export const DateRangePickerInput = forwardRef<{}, 'input'>((_props, ref) => {
+  const {
+    startInputRef,
+    styles,
+    startInputDisplay,
+    handleStartDateChange,
+    displayFormat,
+    fcProps,
+    handleInputBlur,
+    handleInputClick,
+    allowManualInput,
+    labelSeparator,
+    endInputDisplay,
+    endInputRef,
+    handleEndDateChange,
+  } = useDateRangePicker()
+
+  const mergedStartInputRef = useMergeRefs(startInputRef, ref)
+
+  return (
+    <Flex
+      overflowX="auto"
+      sx={{
+        // Hide scrollbars so dual inputs feel like a real normal input.
+        '-ms-overflow-style': 'none',
+        scrollbarWidth: 'none',
+        '&::-webkit-scrollbar': {
+          display: 'none',
+        },
+      }}
+    >
+      <Stack direction="row" align="center">
+        <Input
+          variant="unstyled"
+          aria-label="Start date of range"
+          sx={styles.field}
+          width="6rem"
+          as={ReactInputMask}
+          mask="99/99/9999"
+          value={startInputDisplay}
+          onChange={handleStartDateChange}
+          placeholder={displayFormat.toLowerCase()}
+          maskPlaceholder={displayFormat.toLowerCase()}
```

Comment:
nit: we could move this conversion up so that we don't have so many identical function calls as prop inputs

Reply:
[bd9e907](https://github.com/opengovsg/FormSG/pull/4855/commits/bd9e9077bfd48509c5c94cbc606fcf581a1da17f)